Lib/pdb.py

```diff
@@ -438,7 +439,30 @@ def default(self, line):
         locals = self.curframe_locals
         globals = self.curframe.f_globals
         try:
-            code = compile(line + '\n', '<stdin>', 'single')
+            if (code := codeop.compile_command(line + '\n', '<stdin>', 'single')) is None:
```

Comment:
The problem here is the `\n`. There will be statements that are considered "finish" with the newline character, but aren't without.

For example: ``for i in range(5): print(i)``. Without testing `line + "\n"`, it would be considered as "incomplete" and expecting more. This actually is a test case for pdb and we don't want to break the backward compatibility.